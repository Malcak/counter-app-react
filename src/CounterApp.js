import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  // handleAdd
  const handleAdd = () => {
    // setCounter((c) => c + 1);
    setCounter(counter + 1);
  };

  const handleSubtract = () => {
    setCounter(counter - 1);
  };

  const handleReset = () => {
    setCounter(value);
  };

  return (
    <>
      <h1>CounterApp</h1>
      <h2 id="counterValue">{counter}</h2>

      <button id="add" onClick={handleAdd}>
        +1
      </button>
      <button id="reset" onClick={handleReset}>
        reset
      </button>
      <button id="subtract" onClick={handleSubtract}>
        -1
      </button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number,
};

CounterApp.defaultProps = {
  value: 0,
};

export default CounterApp;
