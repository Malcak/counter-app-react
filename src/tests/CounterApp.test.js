import React from 'react';
import { shallow } from 'enzyme';

import CounterApp from '../CounterApp';

describe('Test in <CounterApp /> component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CounterApp />);
  });
  test('should display the <CounterApp /> component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('should show the default value of 100 sent in in the properties', () => {
    const value = 100;
    const wrapper = shallow(<CounterApp value={value} />);
    const counterValue = wrapper.find('#counterValue').text();
    expect(counterValue).toBe('100');
  });
  test('should increment the counter when the add button is pressed', () => {
    wrapper.find('#add').simulate('click');
    const counterValue = wrapper.find('#counterValue').text();
    expect(counterValue).toBe('1');
  });
  test('should reset the counter when the reset button is pressed', () => {
    const value = 100;
    const wrapper = shallow(<CounterApp value={value} />);
    wrapper.find('#add').simulate('click');
    wrapper.find('#reset').simulate('click');
    const counterValue = wrapper.find('#counterValue').text();
    expect(counterValue).toBe(value.toString());
  });
  test('should decrement the counter when the subtract button is pressed', () => {
    wrapper.find('#subtract').simulate('click');
    const counterValue = wrapper.find('#counterValue').text();
    expect(counterValue).toBe('-1');
  });
});
