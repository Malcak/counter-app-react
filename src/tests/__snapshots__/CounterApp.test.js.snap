// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <CounterApp /> component should display the <CounterApp /> component correctly 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2
    id="counterValue"
  >
    0
  </h2>
  <button
    id="add"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    id="reset"
    onClick={[Function]}
  >
    reset
  </button>
  <button
    id="subtract"
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
